jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -5,16 +6,98 @@ XXX jcssackett 2015-09-18: Licensing for juju.js? It's different then the
 licensing for the GUI.
 */
 
-;
+'use strict';
+
+var module = module;
+
+/**
+ * jujulib provides API access for microservices used by juju.
+ *
+ * jujulib provies access to the APIs for the Juju Environment
+ * Manager (JEM), the juju charmstore, and the juju identity
+ * manager (IdM).
+ */
+(function (exports) {
+
+    /**
+     * Environment object for jujulib.
+     *
+     * Provides access to the JEM API.
+     */
+
+    /**
+     * Initializer
+     *
+     * @function environment
+     * @param url {String} The URL, including scheme and port, of the JEM instance.
+     * @param bakery {Object} A bakery object for communicating with the JEM instance.
+     * @returns {Object} A client object for making JEM API calls.
+     */
+    function environment(url, bakery) {
+        this.jemUrl = url + '/v1';
+        this.bakery = bakery;
+    };
+
+    /**
+     * Wrapper for making requests via the bakery.
+     *
+     * @private _makeRequest
+     * @param path {String} The JEM endpoint to make the request from,
+     *     e.g. '/env'
+     * @param success {function} A callback to be called on success. Takes
+     *     an xhr object as its only parameter.
+     * @param failure {function} A callback to be called on failure. Takes
+     *     an xhr object as its only parameter.
+    */
+    environment.prototype._makeRequest = function(path, success, failure) {
+        this.bakery.sendGetRequest(path, function(xhr) {
+            var data = JSON.parse(xhr.target.responseText);
```

Comment:
Do we know that this responsetext will always be valid? Because this is a lib it might be worth being a little bit more resilient here and having a try/catch around the parse.